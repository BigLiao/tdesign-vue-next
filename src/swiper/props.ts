/* eslint-disable */

/**
 * 该文件为脚本自动生成文件，请勿随意修改。如需修改请联系 PMC
 * updated at 2021-12-05 14:42:17
 * */

import { TdSwiperProps } from './type';
import { PropType } from 'vue';

export default {
  /** 轮播切换动画效果类型 */
  animation: {
    type: String as PropType<TdSwiperProps['animation']>,
    default: 'slide' as TdSwiperProps['animation'],
    validator(val: TdSwiperProps['animation']): boolean {
      return ['slide'].includes(val);
    },
  },
  /** 是否自动播放 */
  autoplay: {
    type: Boolean,
    default: true,
  },
  /** 当前轮播在哪一项（下标） */
  current: {
    type: Number,
  },
  /** 当前轮播在哪一项（下标），非受控属性 */
  defaultCurrent: {
    type: Number,
  },
  /** 轮播滑动方向，包括横向滑动和纵向滑动两个方向 */
  direction: {
    type: String as PropType<TdSwiperProps['direction']>,
    default: 'horizontal' as TdSwiperProps['direction'],
    validator(val: TdSwiperProps['direction']): boolean {
      return ['horizontal', 'vertical'].includes(val);
    },
  },
  /** 滑动动画时长 */
  duration: {
    type: Number,
    default: 300,
  },
  /** 轮播间隔时间 */
  interval: {
    type: Number,
    default: 5000,
  },
  /** 轮播切换时触发 */
  onChange: Function as PropType<TdSwiperProps['onChange']>,
};
