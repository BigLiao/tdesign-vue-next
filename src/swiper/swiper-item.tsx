import { defineComponent } from 'vue';
import { prefix } from '../config';

const name = `${prefix}-swiper-item`;

export default defineComponent({
  name,
  render() {
    const { default: defaultSlot } = this.$slots;
    return (
      <div>
        { defaultSlot && defaultSlot() }
      </div>
    );
  },
});
