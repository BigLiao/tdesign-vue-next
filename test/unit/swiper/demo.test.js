import { mount } from '@vue/test-utils';
import swiper from '@/examples/swiper/demos/swiper.vue';
import vertical from '@/examples/swiper/demos/vertical.vue';

describe('Swiper', () => {
  it('swiper demo works fine', () => {
    const wrapper = mount(swiper);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('vertical demo works fine', () => {
    const wrapper = mount(vertical);
    expect(wrapper.element).toMatchSnapshot();
  });
});
