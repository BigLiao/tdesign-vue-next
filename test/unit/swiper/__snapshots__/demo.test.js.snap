// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Swiper swiper demo works fine 1`] = `
<div
  class="tdesign-demo-block--swiper"
>
  <div
    class="t-swiper"
  >
    <div
      class="t-swiper__content"
    >
      <div
        class="t-swiper__swiper-wrap--horizontal"
        style="width: 400%; transform: translate3d(0%,0,0); transition: transform 0.3s;"
      >
        
        <div>
          
          <div
            class="t-alert t-alert--success"
          >
            <div
              class="t-alert__icon"
            >
              <svg
                class="t-icon t-icon-check-circle-filled"
                version="1.1"
                viewBox="0 0 1024 1024"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M512 96C741.7504576 96 928 282.24954432 928 512 928 741.7504576 741.7504576 928 512 928 282.24954432 928 96 741.7504576 96 512 96 282.24954432 282.24954432 96 512 96ZM690.8655168 353.58818368L464.56401024 579.82818368 328.82684416 444.09785152 283.57201024 489.35268544 464.59134592 670.3720192 509.84617984 625.1171872 736.120352 398.8430176 690.8655168 353.58818368Z"
                  fill-opacity=".9"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
            <div
              class="t-alert__content"
            >
              <!---->
              <div
                class="t-alert__message"
              >
                <div
                  class="t-alert__description"
                >
                  【到期提醒】
                  <!---->
                </div>
                <!---->
              </div>
            </div>
            <!---->
          </div>
          
        </div>
        <div>
          
          <div
            class="t-alert t-alert--info"
          >
            <div
              class="t-alert__icon"
            >
              <svg
                class="t-icon t-icon-info-circle-filled"
                version="1.1"
                viewBox="0 0 1024 1024"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M512 96C741.7504576 96 928 282.24954432 928 512 928 741.7504576 741.7504576 928 512 928 282.24954432 928 96 741.7504576 96 512 96 282.24954432 282.24954432 96 512 96ZM544 416L416 416 416 480 480 480 480 736 608 736 608 672 544 672 544 416ZM544 288L480 288 480 352 544 352 544 288Z"
                  fill-opacity=".9"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
            <div
              class="t-alert__content"
            >
              <!---->
              <div
                class="t-alert__message"
              >
                <div
                  class="t-alert__description"
                >
                  【到期提醒】
                  <!---->
                </div>
                <!---->
              </div>
            </div>
            <!---->
          </div>
          
        </div>
        <div>
          
          <div
            class="t-alert t-alert--warning"
          >
            <div
              class="t-alert__icon"
            >
              <svg
                class="t-icon t-icon-error-circle-filled"
                version="1.1"
                viewBox="0 0 1024 1024"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M512 96C741.7504576 96 928 282.24954432 928 512 928 741.7504576 741.7504576 928 512 928 282.24954432 928 96 741.7504576 96 512 96 282.24954432 282.24954432 96 512 96ZM544 672L480 672 480 736 544 736 544 672ZM544 288L480 288 480 608 544 608 544 288Z"
                  fill-opacity=".9"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
            <div
              class="t-alert__content"
            >
              <!---->
              <div
                class="t-alert__message"
              >
                <div
                  class="t-alert__description"
                >
                  【到期提醒】
                  <!---->
                </div>
                <!---->
              </div>
            </div>
            <!---->
          </div>
          
        </div>
        <div>
          
          <div
            class="t-alert t-alert--error"
          >
            <div
              class="t-alert__icon"
            >
              <svg
                class="t-icon t-icon-error-circle-filled"
                version="1.1"
                viewBox="0 0 1024 1024"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M512 96C741.7504576 96 928 282.24954432 928 512 928 741.7504576 741.7504576 928 512 928 282.24954432 928 96 741.7504576 96 512 96 282.24954432 282.24954432 96 512 96ZM544 672L480 672 480 736 544 736 544 672ZM544 288L480 288 480 608 544 608 544 288Z"
                  fill-opacity=".9"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
            <div
              class="t-alert__content"
            >
              <!---->
              <div
                class="t-alert__message"
              >
                <div
                  class="t-alert__description"
                >
                  【到期提醒】
                  <!---->
                </div>
                <!---->
              </div>
            </div>
            <!---->
          </div>
          
        </div>
        
      </div>
    </div>
    <ul
      class="t-swiper__trigger-wrap"
    >
      
      <li
        class="t-swiper__trigger--active"
      />
      <li
        class=""
      />
      <li
        class=""
      />
      <li
        class=""
      />
      
    </ul>
  </div>
  <div
    class="t-swiper"
  >
    <div
      class="t-swiper__content"
    >
      <div
        class="t-swiper__swiper-wrap--horizontal"
        style="width: 400%; transform: translate3d(0%,0,0); transition: transform 0.5s;"
      >
        
        <div>
          
          <div
            class="t-alert t-alert--success"
          >
            <div
              class="t-alert__icon"
            >
              <svg
                class="t-icon t-icon-check-circle-filled"
                version="1.1"
                viewBox="0 0 1024 1024"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M512 96C741.7504576 96 928 282.24954432 928 512 928 741.7504576 741.7504576 928 512 928 282.24954432 928 96 741.7504576 96 512 96 282.24954432 282.24954432 96 512 96ZM690.8655168 353.58818368L464.56401024 579.82818368 328.82684416 444.09785152 283.57201024 489.35268544 464.59134592 670.3720192 509.84617984 625.1171872 736.120352 398.8430176 690.8655168 353.58818368Z"
                  fill-opacity=".9"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
            <div
              class="t-alert__content"
            >
              <!---->
              <div
                class="t-alert__message"
              >
                <div
                  class="t-alert__description"
                >
                  【到期提醒】
                  <!---->
                </div>
                <!---->
              </div>
            </div>
            <!---->
          </div>
          
        </div>
        <div>
          
          <div
            class="t-alert t-alert--info"
          >
            <div
              class="t-alert__icon"
            >
              <svg
                class="t-icon t-icon-info-circle-filled"
                version="1.1"
                viewBox="0 0 1024 1024"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M512 96C741.7504576 96 928 282.24954432 928 512 928 741.7504576 741.7504576 928 512 928 282.24954432 928 96 741.7504576 96 512 96 282.24954432 282.24954432 96 512 96ZM544 416L416 416 416 480 480 480 480 736 608 736 608 672 544 672 544 416ZM544 288L480 288 480 352 544 352 544 288Z"
                  fill-opacity=".9"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
            <div
              class="t-alert__content"
            >
              <!---->
              <div
                class="t-alert__message"
              >
                <div
                  class="t-alert__description"
                >
                  【到期提醒】
                  <!---->
                </div>
                <!---->
              </div>
            </div>
            <!---->
          </div>
          
        </div>
        <div>
          
          <div
            class="t-alert t-alert--warning"
          >
            <div
              class="t-alert__icon"
            >
              <svg
                class="t-icon t-icon-error-circle-filled"
                version="1.1"
                viewBox="0 0 1024 1024"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M512 96C741.7504576 96 928 282.24954432 928 512 928 741.7504576 741.7504576 928 512 928 282.24954432 928 96 741.7504576 96 512 96 282.24954432 282.24954432 96 512 96ZM544 672L480 672 480 736 544 736 544 672ZM544 288L480 288 480 608 544 608 544 288Z"
                  fill-opacity=".9"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
            <div
              class="t-alert__content"
            >
              <!---->
              <div
                class="t-alert__message"
              >
                <div
                  class="t-alert__description"
                >
                  【到期提醒】
                  <!---->
                </div>
                <!---->
              </div>
            </div>
            <!---->
          </div>
          
        </div>
        <div>
          
          <div
            class="t-alert t-alert--error"
          >
            <div
              class="t-alert__icon"
            >
              <svg
                class="t-icon t-icon-error-circle-filled"
                version="1.1"
                viewBox="0 0 1024 1024"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M512 96C741.7504576 96 928 282.24954432 928 512 928 741.7504576 741.7504576 928 512 928 282.24954432 928 96 741.7504576 96 512 96 282.24954432 282.24954432 96 512 96ZM544 672L480 672 480 736 544 736 544 672ZM544 288L480 288 480 608 544 608 544 288Z"
                  fill-opacity=".9"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
            <div
              class="t-alert__content"
            >
              <!---->
              <div
                class="t-alert__message"
              >
                <div
                  class="t-alert__description"
                >
                  【到期提醒】
                  <!---->
                </div>
                <!---->
              </div>
            </div>
            <!---->
          </div>
          
        </div>
        
      </div>
    </div>
    <ul
      class="t-swiper__trigger-wrap"
    >
      
      <li
        class="t-swiper__trigger--active"
      />
      <li
        class=""
      />
      <li
        class=""
      />
      <li
        class=""
      />
      
    </ul>
  </div>
</div>
`;

exports[`Swiper vertical demo works fine 1`] = `
<div
  class="t-swiper"
>
  <div
    class="t-swiper__content"
  >
    <div
      class="t-swiper__swiper-wrap--vertical"
      style="height: 400%; transform: translate3d(0,0%,0); transition: transform 0.3s;"
    >
      
      <div>
        
        <div
          class="t-alert t-alert--success"
        >
          <div
            class="t-alert__icon"
          >
            <svg
              class="t-icon t-icon-check-circle-filled"
              version="1.1"
              viewBox="0 0 1024 1024"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M512 96C741.7504576 96 928 282.24954432 928 512 928 741.7504576 741.7504576 928 512 928 282.24954432 928 96 741.7504576 96 512 96 282.24954432 282.24954432 96 512 96ZM690.8655168 353.58818368L464.56401024 579.82818368 328.82684416 444.09785152 283.57201024 489.35268544 464.59134592 670.3720192 509.84617984 625.1171872 736.120352 398.8430176 690.8655168 353.58818368Z"
                fill-opacity=".9"
                fill-rule="evenodd"
              />
            </svg>
          </div>
          <div
            class="t-alert__content"
          >
            <!---->
            <div
              class="t-alert__message"
            >
              <div
                class="t-alert__description"
              >
                【到期提醒】
                <!---->
              </div>
              <!---->
            </div>
          </div>
          <!---->
        </div>
        
      </div>
      <div>
        
        <div
          class="t-alert t-alert--info"
        >
          <div
            class="t-alert__icon"
          >
            <svg
              class="t-icon t-icon-info-circle-filled"
              version="1.1"
              viewBox="0 0 1024 1024"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M512 96C741.7504576 96 928 282.24954432 928 512 928 741.7504576 741.7504576 928 512 928 282.24954432 928 96 741.7504576 96 512 96 282.24954432 282.24954432 96 512 96ZM544 416L416 416 416 480 480 480 480 736 608 736 608 672 544 672 544 416ZM544 288L480 288 480 352 544 352 544 288Z"
                fill-opacity=".9"
                fill-rule="evenodd"
              />
            </svg>
          </div>
          <div
            class="t-alert__content"
          >
            <!---->
            <div
              class="t-alert__message"
            >
              <div
                class="t-alert__description"
              >
                【到期提醒】
                <!---->
              </div>
              <!---->
            </div>
          </div>
          <!---->
        </div>
        
      </div>
      <div>
        
        <div
          class="t-alert t-alert--warning"
        >
          <div
            class="t-alert__icon"
          >
            <svg
              class="t-icon t-icon-error-circle-filled"
              version="1.1"
              viewBox="0 0 1024 1024"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M512 96C741.7504576 96 928 282.24954432 928 512 928 741.7504576 741.7504576 928 512 928 282.24954432 928 96 741.7504576 96 512 96 282.24954432 282.24954432 96 512 96ZM544 672L480 672 480 736 544 736 544 672ZM544 288L480 288 480 608 544 608 544 288Z"
                fill-opacity=".9"
                fill-rule="evenodd"
              />
            </svg>
          </div>
          <div
            class="t-alert__content"
          >
            <!---->
            <div
              class="t-alert__message"
            >
              <div
                class="t-alert__description"
              >
                【到期提醒】
                <!---->
              </div>
              <!---->
            </div>
          </div>
          <!---->
        </div>
        
      </div>
      <div>
        
        <div
          class="t-alert t-alert--error"
        >
          <div
            class="t-alert__icon"
          >
            <svg
              class="t-icon t-icon-error-circle-filled"
              version="1.1"
              viewBox="0 0 1024 1024"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M512 96C741.7504576 96 928 282.24954432 928 512 928 741.7504576 741.7504576 928 512 928 282.24954432 928 96 741.7504576 96 512 96 282.24954432 282.24954432 96 512 96ZM544 672L480 672 480 736 544 736 544 672ZM544 288L480 288 480 608 544 608 544 288Z"
                fill-opacity=".9"
                fill-rule="evenodd"
              />
            </svg>
          </div>
          <div
            class="t-alert__content"
          >
            <!---->
            <div
              class="t-alert__message"
            >
              <div
                class="t-alert__description"
              >
                【到期提醒】
                <!---->
              </div>
              <!---->
            </div>
          </div>
          <!---->
        </div>
        
      </div>
      
    </div>
  </div>
  <ul
    class="t-swiper__trigger-wrap"
  >
    
    <li
      class="t-swiper__trigger--active"
    />
    <li
      class=""
    />
    <li
      class=""
    />
    <li
      class=""
    />
    
  </ul>
</div>
`;
